src/v/iceberg/catalog.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "base/seastarx.h"
+#include "iceberg/partition.h"
+#include "iceberg/schema.h"
+#include "iceberg/transaction.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/sstring.hh>
+
+namespace iceberg {
+
+struct table_identifier {
+    ss::sstring ns;
```

Comment:
Since namespaces can be nested, and [nested namespaces should be joined by unit separator](https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L1743-L1757); should this field be a collection? If we join the nesting components early then it might be harder to log etc, we might have to split for logging.

If we keep a collection here then the lower level implementation can produce the joined string right before making api calls to the catalog as late as possible.